backend/src/main/java/com/redhat/cloud/notifications/events/EventConsumer.java

```diff
@@ -163,10 +167,10 @@ public void init() {
                          */
                         LOGGER.infof(throwable, "Could not process the payload: %s", payload);
                     }
-                    /*
-                     * This counter main purpose is to prevent a test race condition.
-                     */
-                    consumedCounter.increment();
+                    // bundleName[0] and appName[0] are null when the action parsing failed.
+                    String bundle = bundleName[0] == null ? "" : bundleName[0];
+                    String application = appName[0] == null ? "" : appName[0];
+                    consumedTimer.stop(registry.timer(CONSUMED_TIMER_NAME, "bundle", bundle, "application", application));
```

Comment:
is this tested anywhere?

Reply:
Good question, the answer is no but I'll add something.

Reply:
Done.